src/internal_modules/roc_rtp/link_meter.h

```diff
@@ -25,7 +29,7 @@
 namespace roc {
 namespace rtp {
 
-//! RTP link meter.
+//! Link meter.
```

Comment:
> I see that currently read() method updated recovered_packets metric. However, link meter is not actually used as reader, so that metric doesn't work?

It actually does! In `ReceiverSession` ctor:

```
pkt_reader = timestamp_injector_.get();

source_meter_->set_reader(*pkt_reader);
pkt_reader = source_meter_.get();

// Third part of pipeline: chained frame readers from depacketizer to mixer.
// Mixed reads frames from this pipeline, and in the end it requests packets
// from packet readers pipeline.
audio::IFrameReader* frm_reader = NULL;
```

Reply:
Oh, I see. But see my other comment. Since currently read() only updates recovered_packets metric, and I think depacketizer is a better place to compute this metric, we can simplify link meter and remove read().

If you're not using recovered_packets in code of course, but I think no?

Reply:
I'd like to keep recover_packets in Metrics because it is a metric. It is not being used now, although this value can be useful to tell the difference between congestion vs arbitrary loss.

Reply:
I think you misunderstood me, what I suggest is to keep it being a metric, but move it from link meter metrics to depacketizer metrics.

Depacketizer does not have metrics right now, but we'll add it in #681. Only depacketizer can compute "late packets" metric, so it has to have metrics in either case. And it also *could* compute "recovered packets" metrics and it's a natural place to do it (since depacketizer is placed right after fec reader). So, why not to compute "recovered packets" metric in depacketizer and make link meter simpler by removing read?

So my suggestion was, if "recovered packets" is not used *in this PR*, remove it's computation from link meter, and then re-add it to depacketizer in #681.

Reply:
I'm sorry, it `recovered_packets` metric is used in pipeline tests, and I recall these tests motivated me to add it

Reply:
I've implemented depacketizer metrics:

```cpp
struct DepacketizerMetrics {
    //! Cumulative count of packets from which we decoded samples.
    //! Incremented each time depacketizer starts decoding a packet.
    uint64_t decoded_packet_count;

    //! Cumulative count of packets dropped because they were late.
    //! Incremented each time depacketizer drops a packet.
    uint64_t late_packet_count;

    //! Cumulative count of packets repaired by FEC.
    //! Incremented each time depacketizer reads a packet with FlagRestored.
    //! This metric excludes late packets that were repaired but then dropped.
    uint64_t recovered_packet_count;
};
```

And added them to pipeline metrics:

```cpp
struct ReceiverParticipantMetrics {
    //! Link metrics.
    packet::LinkMetrics link;

    //! Latency metrics.
    audio::LatencyMetrics latency;

    //! Depacketizer metrics.
    audio::DepacketizerMetrics depacketizer;

    ReceiverParticipantMetrics() {
    }
};
```

If you want, you can try to switch API to using recovered_packet_count from depacketizer instead of link meter, which should allow you to fix the test. Otherwise we can try it later, then please add this to the issue.

Reply:
My bad, it's actually not needed for the test